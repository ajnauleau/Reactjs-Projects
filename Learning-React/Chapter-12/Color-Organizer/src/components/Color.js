import React from 'react';
import StarRating from "./StarRating";
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
const { Component } = React;

export class Color extends Component {

    render() {
        const { id, title, color, rating, onRemove, onRate, history} = this.props
        return (
            <section className="color" style={this.style}>
                <h1 ref="title"
                    onClick={() => history.push(`/${id}`)}>{title}</h1>
                <button onClick={onRemove}>
                    X
                </button>
                <div className="color"
                     onClick={() => history.push(`/${id}`)}
                     style={{ backgroundColor: color }}>
                       <p></p>
                </div>
                <div>
                    <StarRating starsSelected={rating} onRate={onRate}/>
                </div>
            </section>
        )
    }

}

Color.propTypes = {
    title: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    rating: PropTypes.number,
    onRemove: PropTypes.func,
    onRate: PropTypes.func
}

Color.defaultProps = {
    rating: 0,
    onRemove: f=>f,
    onRate: f=>f
}

export default withRouter(Color);
