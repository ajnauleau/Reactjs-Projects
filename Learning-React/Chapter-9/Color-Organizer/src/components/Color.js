import React from 'react';
import {StarRating} from "./StarRating";
import {rateColor, removeColor} from "../actions";
import { PropTypes } from 'prop-types';
const { Component } = React;

export class Color extends Component {

    render() {
        const { store } = this.context
        const { id, title, color, rating } = this.props;
        return (
            <section className="color" style={this.style}>
                <h1 ref="title">{title}</h1>
                <button onClick={() =>
                    store.dispatch(
                        removeColor(id)
                    )
                }>X</button>
                <div className="color"
                     style={{ backgroundColor: color }}>
                    <p></p>
                </div>
                <div>
                    <StarRating starsSelected={rating}
                                onRate={rating =>
                                    store.dispatch(
                                        rateColor(id, rating)
                                        )
                                    } />
                </div>
            </section>
        )
    }
}

Color.contextTypes = {
    store: PropTypes.object
}
