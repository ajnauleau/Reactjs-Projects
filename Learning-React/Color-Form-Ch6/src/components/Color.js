import React from 'react';
import {StarRating} from "./StarRating";
const { Component } = React;

export const Color = ({ title, color, rating=0, onRemove=f=>f, onRate=f=>f }) =>
    <section className="color">
        <h1>{title}</h1>
        <button onClick={onRemove}>X</button>
        <div className="color"
            style={{ backgroundColor: color }}>
            <p>Color</p>
        </div>
        <div>
            <StarRating starsSelected={rating} onRate={onRate} />
        </div>
    </section>
