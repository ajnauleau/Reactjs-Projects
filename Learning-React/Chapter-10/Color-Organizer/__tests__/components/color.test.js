
import toJSON from 'enzyme-to-json'
import { compose } from 'redux'
import Color from '../../src/components/Color'

const { shallow, mount } = Enzyme

describe("<Color /> UI Component", () => {

    const shallowExpect = compose(expect,toJSON,shallow)

    it("Renders correct properties", () =>
        shallowExpect(
            <Color title="Test Color" color="#F0F0F0" rating={3}  timestamp="Mon Apr 11 2016 12:54:19 GMT-0700 (PDT)" />
        ).toMatchSnapshot()
    )

    it("Invokes onRate property", () => {
        mount(<Color title="Test Color" color="#F0F0F0" />)
            .find("div.star-container")
            .childAt(2)
            .simulate('click')
    })

    it("Invokes onRate property", () => {
        mount(<Color title="Test Color" color="#F0F0F0" />)
            .find("button")
            .simulate('click')
    })

    it("Invokes onRate property", () => {
        mount(<Color title="Test Color" color="#F0F0F0" />)
            .find("div.star-rating")
    })

})
