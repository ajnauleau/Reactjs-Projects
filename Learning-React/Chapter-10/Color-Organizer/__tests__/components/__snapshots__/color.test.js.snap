// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Color /> UI Component Renders correct properties 1`] = `
<section
  className="color"
>
  <h1>
    Test Color
  </h1>
  <button
    onClick={[Function]}
  >
    X
  </button>
  <div
    className="color"
    style={
      Object {
        "backgroundColor": "#F0F0F0",
      }
    }
  >
    <p />
  </div>
  <div>
    <StarRating
      onRate={[Function]}
      starsSelected={3}
    />
  </div>
</section>
`;
